File: Coding-Challenges/prefixMapSum/prefixMapSum.js
#!/usr/bin/env node

const assert = require('assert')

class PrefixMapSum {
  constructor() {

  }

  insert(key, value) {

  }

  sum(prefix) {

  }
}

const mapSum = new PrefixMapSum()
mapSum.insert("columnar", 3)
assert.strictEqual(mapSum.sum("col"), 3)
mapSum.insert("column", 2)
assert.strictEqual(mapSum.sum("col"), 5)

File: Coding-Challenges/prefixMapSum/prefix_map_sum.py
class PrefixMapSum:

    def __init__(self):
        pass

    def insert(self, key: str, value: int):
        pass

    def sum(self, prefix: str):
        pass


map_sum = PrefixMapSum()
map_sum.insert("columnar", 3)
assert map_sum.sum("col") == 3
map_sum.insert("column", 2)
assert map_sum.sum("col") == 5
